sql/ddl_internal.sql

```diff
@@ -119,10 +119,16 @@ BEGIN
 
     IF num_slices IS NULL THEN
         -- Open dimension
-        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
+        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'DATE', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
             RAISE EXCEPTION 'illegal type for column "%": %', column_name, column_type
             USING ERRCODE = 'IO102';
         END IF;
+        IF column_type = 'DATE'::regtype AND 
+            (interval_length <= 0 OR interval_length % _timescaledb_internal.interval_to_usec('1 day') != 0) 
```

Comment:
Still not clear to me why we are using usec internally for DATE. Was there a decision made here? Maybe not a big deal but would like to know.

Reply:
Basically this is for consistency. Since intervals can be specified as an integer type in create_hypertable, I thought it's much more clear if we keep consistent units. Also think it's better for internal code as well.

Reply:
If you have an integer time type, the unit is whatever the user makes it (we cannot know if they are doing, e.g.,  seconds, milliseconds, or hours). So, in a way the consistent way of doing this is to use whatever unit that the user has specified, which in this case is days.  With your approach, DATE is the sole exception that does not use its "native" unit.

Reply:
So I think there are 2 cases:

1) Cases where we know the time unit -TIMESTAMP, TIMESTAMPTZ, DATE
2) Cases where we don't know the time unit - all integer cases

I'd rather keep all the units under (1) as microseconds. I don't see a reason not to. Otherwise all functions that accept intervals or time values (for example, drop_chunks, create_hypertable as introduced in #181, etc) will have to be special cased. Note that the full range of DATE types can still be supported.